delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeViewsCoordinator.scala

```diff
@@ -103,6 +86,34 @@ object CompositeViewsCoordinator {
   private val metadata: ProjectionMetadata = ProjectionMetadata("system", "composite-views-coordinator", None, None)
   private val logger: Logger               = Logger[CompositeViewsCoordinator]
 
+  def cleanupCurrent(
+      cache: KeyValueStore[ViewRef, ActiveViewDef],
+      viewDef: CompositeViewDef,
+      destroy: ActiveViewDef => Task[Unit]
+  ): Task[Unit] = {
+    val ref = viewDef.ref
+    cache.get(ref).flatMap { cachedOpt =>
+      (cachedOpt, viewDef) match {
+        case (Some(cached), active: ActiveViewDef) if cached.projection == active.projection =>
+          Task.delay(logger.info(s"Projection '${cached.projection}' is already running and will not be recreated."))
+        case (Some(cached), _: ActiveViewDef)                                                              =>
+          Task.delay(
+            logger.info(s"View '${ref.project}/${ref.viewId}' has been updated, cleaning up the current one.")
+          ) >>
+            destroy(cached)
+        case (Some(cached), _: DeprecatedViewDef)                                                          =>
+          Task.delay(
+            logger.info(s"View '${ref.project}/${ref.viewId}' has been deprecated, cleaning up the current one.")
+          ) >>
+            destroy(cached)
+        case (None, _)                                                                                     =>
```

Comment:
Probably out-of-scope but we are missing this now in the ES and BG coordinators

Reply:
Ok, we can have a look at that in another PR